### ISSUE TITLE
the accuracy of imagenet decreases a lot🤔[question] 

### ISSUE THREAD

**Issue by YUjunuuuuu** [determined-ai/determined#5661]
### Describe your question

hello
In my experiment, I do not use the pre-trained weights. But the accuracy of the imagenet decreases a lot. Can you give me some advice? In local test(trained on  a machine with 8 gpus), the top1 accuracy is 73%. However, when trained on (4 machines and each machine with 8 gpus, total 32 gpus), the accuracy is only 58%. The learning rate(0.1*32) is set to the linear  rule and gobal batch szie is 256 * 32. Epoch is set as 100.

**Comment by ioga** [comment]
hello, we support hyperparameter search: https://docs.determined.ai/latest/reference/reference-training/experiment-config-reference.html#hyperparameters
you can setup parameter search ranges for your global batch size and learning rate, and see how it behaves and where it performs the best.

**Comment by liamcli** [comment]
what batch size did you use with 8 gpus to reach 73% accuracy?  1024? 

agree with ioga that it could be worth doing some hyperparameter tuning of learning rate & batch size.

**Comment by YUjunuuuuu** [comment]
batch size=2048
learning rate=0.1(epoch=100,with lr decrease 0.1 each 30 epochs)

**Comment by YUjunuuuuu** [comment]
can you give me a suggested value?

**Comment by liamcli** [comment]
if you used lr=0.1 and batch_size=2048 with 8 gpus, linear scaling with 32 gpus would give lr=0.4 and batch_size=8192.  i'd also suggest lr annealing and lr warmup if you are not using that already.  especially with the learning lr for 32 gpus.

**Comment by YUjunuuuuu** [comment]
ok,thank you

**Comment by YUjunuuuuu** [comment]
@ioga hi, the definition of top1 accuracy is copied from the provided file(imagenet_pytorch.tgz).Whether should i define a metricReducer to caculate the final accuracy?

**Comment by ioga** [comment]
no, I don't believe you'd need a custom metric reducer.

**Comment by YUjunuuuuu** [comment]
@ioga hi
The accuracy between your provided file(imagenet_pytorch.tgz) and my local test(We load the weights which trained on the determined framework) is different(68% vs 75%). Whether the defination of top1 accuracy has some error ? Can you give me some  advice, thank you

**Comment by garrett361** [comment]
Hi @YUjunuuuuu , can you describe your setup in greater detail?  I don't think we can give useful advice without more information.

My understanding is that with the [imagenet_pytorch code](https://github.com/determined-ai/determined/tree/master/examples/tutorials/imagenet_pytorch) you are achieving 68% accuracy and you are then comparing this result to other code which gives 75% accuracy. Can you clarify what factors are changed in the two tests?
It sounds like you are using the same model weights in both tests, the 68% and 75% ones, and the same code for computing [accuracy](https://github.com/determined-ai/determined/blob/b69bab1c1c59172963ec0f8eb2caba6cc6c24c1a/examples/tutorials/imagenet_pytorch/model_def.py#L160). Is that right?

**Comment by YUjunuuuuu** [comment]
Hi, you are right .  The imagenet_pytorch code is only modified some params in distribute.yaml. The checkpoint(state_dict.pth) is the same in both tests.what bother me is that the same ckpt(state_dict.pth) in both tests gives different accuracy.
For detail, I use determined to train the model. This will save a state_dict.pth.But when I load the checkpoint in my local test(give a test.py and calculate the accuracy), the accuracy in two tests  are different.
The two tests are below.
**determined code:**
![eafe7a3df497615e3b8de9ff5e251f5](https://user-images.githubusercontent.com/49989327/211700336-243ea9c2-29c5-45f6-87a9-786e6401527b.png)
**local test code:**
<pre>
import torch
import torchvision
import torchvision.transforms as transforms
import torchvision.datasets as datasets
import torchvision.models as models

import time

arch = "resnet50"
batch_size = 512
valdir = "/data/14e9aaec-a625-4721-83b2-137ad3671fac/Imagenet2012/val/"

class AverageMeter(object):
    def __init__(self):
        self.reset()
    def reset(self):
        self.val = 0
        self.avg = 0
        self.sum = 0
        self.count = 0
    def update(self, val, n=1):
        self.val = val
        self.sum += val * n
        self.count += n
        self.avg = self.sum / self.count
def accuracy(output, target, topk=(1,)):
    """Computes the precision@k for the specified values of k"""
    maxk = max(topk)
    batch_size = target.size(0)
    _, pred = output.topk(maxk, 1, largest=True, sorted=True)
    pred = pred.t()
    correct = pred.eq(target.reshape(1, -1).expand_as(pred))
    res = []
    for k in topk:
        correct_k = correct[:k].reshape(-1).float().sum(0, keepdim=True)
        res.append(correct_k.mul_(100.0 / batch_size))
    return res
normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
val_dataset = datasets.ImageFolder(
    valdir,
    transforms.Compose([
        transforms.Resize(256),
        transforms.CenterCrop(224),
        transforms.ToTensor(),
        normalize
    ])
)
val_loader = torch.utils.data.DataLoader(
    val_dataset,
    batch_size=batch_size,
    shuffle=False,
    num_workers=4,
    pin_memory=True,
)
print("=> creating model '{}'".format(arch))
model = models.__dict__[arch]().cuda()
state_dict = torch.load("state_dict.pth", map_location="cpu")["'models_state_dict'"]
model.load_state_dict(state_dict[0])
model.eval()
v_s = time.time()
print("--------val-step--------")
_losses = AverageMeter()
_top1 = AverageMeter()
_top5 = AverageMeter()
# switch to evaluate mode
model.eval()
correct = 0.0
for j, (_input, _target) in enumerate(val_loader):
    _target = _target.cuda()
    _input = _input.cuda()
    with torch.no_grad():
        # compute output
        _output = model(_input)
        # measure accuracy and record loss
        pred = _output.data.max(1)[1]
        correct += pred.eq(_target.data).sum()
        _prec1, _prec5 = accuracy(_output.data, _target, topk=(1, 5))
        _top1.update(_prec1[0], _input.size(0))
        _top5.update(_prec5[0], _input.size(0))
acc = correct / len(val_loader.dataset)
v_e = time.time()
print("acc: {}".format(acc))
print(' * Prec@1 {top1.avg:.3f} Prec@5 {top5.avg:.3f}'.format(top1=_top1, top5=_top5))
print("val cost {} seconds".format(v_e - v_s))
<pre/>

**Comment by liamcli** [comment]
It looks like the validation dataloader is using incorrect transforms in our example:
```            
val_dataset = datasets.ImageFolder(                                                     
    traindir,                                                                           
    transforms.Compose(                                                                 
        [                                                                               
            transforms.RandomResizedCrop(224),                                          
            transforms.RandomHorizontalFlip(),                                          
            transforms.ToTensor(),                                                      
            self.normalize,                                                             
        ]                                                                               
    ),                                                                                  
)  
```

the accuracy should be closer to expected if you remove the random crop and flip.  please try that and let me know if that doesn't help.

thanks for reporting this issue!  we'll fix it for the next release.

**Comment by YUjunuuuuu** [comment]
Thank you, it works. The validation dataloader in two tests are different. When I remove the random crop and flip, the accuracy in two tests are equal.
